crap-advisor-frontend/src/core/components/RestroomMarker/index.tsx

```diff
@@ -1,27 +1,44 @@
 import React from "react";
-import {Marker, Tooltip} from "react-leaflet";
+import {Marker, Tooltip, useMap} from "react-leaflet";
 import L from "leaflet";
-import restroomImage from "../../../assets/restroomMarker.png";
 import "./index.css";
 import {Restroom} from "../../../app/typings";
 import {IMapProps} from "../Map";
 import {LeafletEvent, LatLngExpression} from 'leaflet'
 
+import restroom1 from "../../../assets/restroom_2.png";
+import restroom2 from "../../../assets/restroom_2.png";
+import restroom3 from "../../../assets/restroom_4.png";
+import restroom4 from "../../../assets/restroom_4.png";
+import restroom5 from "../../../assets/restroom_5.png";
+
 interface IRestroomMakerProps extends Pick<IMapProps, 'selectRestroom'> {
-    restroom: Restroom;
+    restroom: Restroom
 }
 
 const RestroomMarker = ({restroom, selectRestroom}: IRestroomMakerProps) => {
+
+    const iconImages = [restroom1, restroom2, restroom3, restroom4, restroom5]
+    let currentImage = restroom4
```

Comment:
I'm not sure that is a good realisation. Worth a look at the useState hook, where state will be is currentImage variable.

Reply:
Ok, I think i need to deep dive into those so-called "hooks"